packages/git/src/browser/dirty-diff/dirty-diff-manager.ts

```diff
@@ -136,6 +139,18 @@ export class DirtyDiffManager {
         return undefined;
     }
 
+    protected async canHandle(uri: string): Promise<boolean> {
+        if (new URI(uri).scheme !== 'file') {
+            return false;
+        }
+        const repository = this.repositoryTracker.selectedRepository;
+        if (repository === undefined) {
+            return false;
+        }
+        const errorUnmatch = true;
+        return this.git.lsFiles(repository, uri, { errorUnmatch });
```

Comment:
I think that will break some cases. I'd rather push that check down to the models, but let them create for the opened editor: 
https://github.com/theia-ide/theia/blob/9594d361045edb253232177d6d3e60e596f6dc3b/packages/git/src/browser/dirty-diff/dirty-diff-manager.ts#L228-L232. 

A file which is not version controlled can be opened in the editor, then you can `git add` it to stage, and then modify it. Current implementation creates models for all opened `file` editors. FYI, in #1526, there will be more improvements regarding reading of resources on demand.

Reply:
That is a good idea. Do you have any objections changing the method to `async`?

Reply:
All right.